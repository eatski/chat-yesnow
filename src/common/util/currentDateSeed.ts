const date = Date.now();
export const CURRENT_HOUR_SEED = Math.floor(
	date / (1000 * 60 * 60 * 12),
).toString();
