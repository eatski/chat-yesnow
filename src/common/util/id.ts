import { generate } from "randomstring"

export const generateId = () => {
    return generate(15);
}