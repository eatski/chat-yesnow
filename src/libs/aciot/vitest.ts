import { AxiosInstance } from "axios";
import { AciotCore, CacheMode } from "./core";
import { afterAll, beforeAll } from "vitest";
import path from "path";
import filenamify from 'filenamify';

export const applyTestHooks = (
	axios: AxiosInstance,
	config: {
		mode: CacheMode;
	},
) => {
	let aciot: AciotCore | null;
	beforeAll(({ name, file }) => {
		if (!file?.name) {
			throw new Error("suite.file.name is not defined");
		}
		const parsedPath = path.parse(file.name);
		const filename = filenamify(name, { replacement: "_" });
		aciot = new AciotCore({
			cacheBasePath: path.resolve(
				process.cwd(),
				parsedPath.dir,
				"__data__",
				parsedPath.name + parsedPath.ext,
				filename
			),
		});
		aciot.applyInterceptors(axios, config.mode);
	});
	afterAll(async ({ name, tasks }) => {
		if (tasks.every((t) => t.result?.state === "pass")) {
			console.info(name, "Clearing unused cache...");
			await aciot?.clearUnusedCache();
		}
	});
};
