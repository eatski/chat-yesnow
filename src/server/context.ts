import { authConfig } from "@/pages/api/auth/[...nextauth]";
import { TRPCError } from "@trpc/server";
import { CreateNextContextOptions } from "@trpc/server/adapters/next";
import { getServerSession } from "next-auth/next";
import { setTimeout } from "timers/promises";
import { verifyRecaptcha } from "./services/recaptcha";
import {
	AB_TESTING_COOKIE_NAME,
	AB_TESTING_VARIANTS,
	getAorBRandom,
	validateABTestingVariant,
} from "@/common/abtesting";
import { get } from "@vercel/edge-config";

export const createContext = async (context: CreateNextContextOptions) => {
	return {
		setupABTestingVariant: async () => {
			const abTestRate = await get("abTestRate");
			if (
				!abTestRate ||
				!(typeof abTestRate === "number") ||
				isNaN(abTestRate) ||
				abTestRate < 0 ||
				abTestRate > 1
			) {
				return AB_TESTING_VARIANTS.ONLY_SONNET;
			}
			const cookieValue = context.req.cookies[AB_TESTING_COOKIE_NAME];
			const validated = cookieValue && validateABTestingVariant(cookieValue);
			if (validated) {
				return validated;
			}
			const random = getAorBRandom(abTestRate);
			context.res.setHeader(
				"Set-Cookie",
				`${AB_TESTING_COOKIE_NAME}=${random}; Path=/; HttpOnly; Secure; SameSite=Strict`,
			);
			return random;
		},
		getUserOptional: async () => {
			const session = await getServerSession(
				context.req,
				context.res,
				authConfig,
			);
			return session?.user?.email && session.custom?.userId
				? {
						id: session.custom.userId,
						email: session.user.email,
				  }
				: null;
		},
		getUser: async () => {
			const session = await getServerSession(
				context.req,
				context.res,
				authConfig,
			);
			if (!session?.user?.email || !session.custom?.userId) {
				throw new TRPCError({
					code: "UNAUTHORIZED",
				});
			}
			return {
				id: session.custom.userId,
				email: session.user.email,
			};
		},
		doRevalidate(url: string) {
			const retryable = (count: number, error?: any): Promise<void> => {
				if (count) {
					return setTimeout(3000).then(() =>
						context.res.revalidate(url).catch((e) => {
							return retryable(count - 1, e);
						}),
					);
				}
				console.error(error);
				return Promise.reject(
					new Error("revalidation failed", {
						cause: error,
					}),
				);
			};
			return retryable(10);
		},
		verifyRecaptcha(token: string) {
			return verifyRecaptcha(token).catch((e) => {
				throw new TRPCError({
					code: "BAD_REQUEST",
					cause: e,
				});
			});
		},
		isThankYouUser: () =>
			context.req.cookies.thankyou === process.env.THANKYOU_CODE,
	};
};
