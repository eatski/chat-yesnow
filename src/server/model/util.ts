export type Public<T, F extends keyof T> = {
	[K in F]: DateToNumber<T[K]>;
} & {
	[K in Exclude<keyof T, F>]?: never;
};

export type Override<
	T extends Record<keyof U, unknown>,
	U extends Partial<Record<keyof T, unknown>>,
> = Omit<T, keyof U> & U;

type DateToNumber<T> = T extends Date ? number : T;
