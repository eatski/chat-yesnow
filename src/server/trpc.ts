import { authConfig } from '@/pages/api/auth/[...nextauth]';
import { initTRPC, TRPCError } from '@trpc/server';
import { CreateNextContextOptions } from '@trpc/server/adapters/next';
import { getServerSession } from 'next-auth/next';
import { setTimeout } from 'timers/promises';

export const createContext = async (context: CreateNextContextOptions) => {
    const session = await getServerSession(context.req, context.res, authConfig);
    if(!session || !session.user?.email){
        throw new TRPCError({
            code: 'UNAUTHORIZED',
        })
    }
    return {
      user: {
        email: session.user.email,
      },
      doRevalidate(url: string){
        const retryable = (count: number,error?: any): Promise<void> => {
          if(count){
            return setTimeout(3000).then(() => context.res.revalidate(url).catch((e) => {
              return retryable(count - 1,e)
            }))
          }
          console.error(error);
          return Promise.reject( new Error("revalidation failed",{
            cause: error
          }))
        }
        return retryable(10)
      },
    };
};

/**
 * Initialization of tRPC backend
 * Should be done only once per backend!
 */
const t = initTRPC.context<typeof createContext>().create();

/**
 * Export reusable router and procedure helpers
 * that can be used throughout the router
 */
export const router = t.router;
export const procedure = t.procedure;