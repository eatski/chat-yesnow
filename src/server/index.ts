import { hello } from './procedures/hello';
import { post } from './procedures/post';
import { put } from './procedures/put';
import { question } from './procedures/question';
import { truth } from './procedures/truth';
import { router } from './trpc';

export const appRouter = router({
  hello,
  question,
  truth,
  post,
  put
});

// Export type router type signature,
// NOT the router itself.
export type AppRouter = typeof appRouter;