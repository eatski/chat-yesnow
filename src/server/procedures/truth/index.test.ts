import { describe, test, expect, beforeAll } from "vitest";
import { appRouter } from "@/server";
import { resolve } from "path";
import { prepareStoryFromYaml } from "@/test/prepareStory";
import { generateId } from "@/common/util/id";
import { applyTestHooks } from "@/libs/msw-cache/vitest";
const never = () => {
	throw new Error("Never");
};

const TEST_USER1 = {
	id: generateId(),
	email: "yesonor@example.com",
};

const TEST_USER2 = {
	id: generateId(),
	email: "not-yesonor@example.com",
};

describe("trpc/truth", () => {
	const testYamlPath1 = resolve(process.cwd(), "fixtures", "test.yaml");
	const testYamlPath2 = resolve(process.cwd(), "fixtures", "test2.yaml");
	const TEST1_ID = generateId();
	const TEST2_ID = generateId();
	const TEST_ID_PRIVATE = generateId();
	applyTestHooks();
	beforeAll(async () => {
		console.log(TEST1_ID, TEST2_ID, TEST_ID_PRIVATE);
		const cleanup1 = await prepareStoryFromYaml(testYamlPath1, {
			storyId: TEST1_ID,
			authorId: TEST_USER1.id,
			published: true,
		});
		const cleanup2 = await prepareStoryFromYaml(testYamlPath2, {
			storyId: TEST2_ID,
			authorId: TEST_USER1.id,
			published: true,
		});
		const cleanup3 = await prepareStoryFromYaml(testYamlPath1, {
			storyId: TEST_ID_PRIVATE,
			authorId: TEST_USER2.id,
			published: false,
		});
		return async () => {
			await cleanup1();
			await cleanup2();
			await cleanup3();
		};
	});
	const testee = appRouter.createCaller({
		getUserOptional: async () => null,
		getUser: never,
		doRevalidate: never,
		verifyRecaptcha: () => Promise.resolve(),
		isDeveloper: () => false,
	});
	describe("解答した内容に対して、結果が返る", () => {
		test.each([
			{
				storyId: TEST1_ID,
				text: "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
			},
			{
				storyId: TEST1_ID,
				text: "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんは運転手なので、終点の後も乗っていた。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんは運転手であり、終点についてもずっと運転席に座っている。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんはバスの運転手なので運転席から離れない。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんは運転手なので、終点の後も乗っていた。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんは市営バスの運転手なので、終点の後も乗っている。",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんはバスの運転手だった",
			},
		])(
			"真相に対して正しい解答をするとCoversが返る。 [$text]",
			async ({ text, storyId }) => {
				const result = await testee.truth({
					storyId,
					text,
					recaptchaToken: "anytoken",
				});
				expect(result.result).toEqual("Covers");
				expect(result).toMatchSnapshot();
			},
		);
		test.each([
			{
				storyId: TEST1_ID,
				text: "山田さんは人を殺害し、見つからないように女性トイレに隠れた",
			},
			{
				storyId: TEST1_ID,
				text: "山田さんは覗きをするために女性用トイレに入った。",
			},
			{
				storyId: TEST1_ID,
				text: "山田さんは女性になりたくて女性用トイレに入った",
			},
			{
				storyId: TEST1_ID,
				text: "山田さんは女性用トイレに入った",
			},
			{
				storyId: TEST1_ID,
				text: "山田さんは隠れるために女性用トイレに入った",
			},
			{
				storyId: TEST2_ID,
				text: "花子さんは死んでしまったため、終点でもずっと座っていた。",
			},
		])(
			"真相に対して間違えた解答をするとがWrong返る [$text]",
			async ({ text, storyId }) => {
				const result = await testee.truth({
					storyId,
					text,
					recaptchaToken: "anytoken",
				});
				expect(result.result).toEqual("Wrong");
				expect(result).toMatchSnapshot();
			},
		);
	});
	describe("storyの参照権限", () => {
		test.each([TEST_USER1, TEST_USER2, null])(
			"publicなstoryに対して質問すると回答が返ってくる",
			async (user) => {
				const testee = appRouter.createCaller({
					getUserOptional: async () => user,
					getUser: never,
					doRevalidate: never,
					verifyRecaptcha: () => Promise.resolve(),
					isDeveloper: () => false,
				});
				const text =
					"山田さんは人を殺害し、男性用トイレに隠そうとしていたが、見つかりそうになり女性用トイレに逃げた。";
				const result = await testee.truth({
					storyId: TEST1_ID,
					text,
					recaptchaToken: "anytoken",
				});
				expect(result.result).toBeDefined();
			},
		);
		test.each([TEST_USER1, null])(
			"privateなstoryに対して質問すると回答が返ってこない",
			async (user) => {
				const testee = appRouter.createCaller({
					getUserOptional: async () => user,
					getUser: never,
					doRevalidate: never,
					verifyRecaptcha: () => Promise.resolve(),
					isDeveloper: () => false,
				});
				const text =
					"山田さんは人を殺害し、男性用トイレに隠そうとしていたが、見つかりそうになり女性用トイレに逃げた。";
				expect(
					testee.truth({
						storyId: TEST_ID_PRIVATE,
						text,
						recaptchaToken: "anytoken",
					}),
				).rejects.toMatchInlineSnapshot("[TRPCError: NOT_FOUND]");
			},
		);
		test("自分の作成したstoryの場合、privateなstoryに対して質問しても回答が返ってくる", async () => {
			const testee = appRouter.createCaller({
				getUserOptional: async () => TEST_USER2,
				getUser: never,
				doRevalidate: never,
				verifyRecaptcha: () => Promise.resolve(),
				isDeveloper: () => false,
			});
			const text =
				"山田さんは人を殺害し、男性用トイレに隠そうとしていたが、見つかりそうになり女性用トイレに逃げた。";
			const result = await testee.truth({
				storyId: TEST_ID_PRIVATE,
				text,
				recaptchaToken: "anytoken",
			});
			expect(result.result).toBeDefined();
		});
	});
});
