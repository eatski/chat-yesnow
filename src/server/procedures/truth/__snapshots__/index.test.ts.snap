// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに…'] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。'] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['花子さんは運転手であり、終点についても、ずっと運転席に座っている。'] 1`] = `
{
  "distance": 0.3,
  "input": "花子さんは運転手であり、終点についても、ずっと運転席に座っている。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['花子さんは運転手であり、終点についてもずっと運転席に座っている。'] 1`] = `
{
  "distance": 0.3,
  "input": "花子さんは運転手であり、終点についてもずっと運転席に座っている。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['花子さんは運転手であり、終点の後もずっと運転席に座っている。'] 1`] = `
{
  "distance": 0.31,
  "input": "花子さんは運転手であり、終点の後もずっと運転席に座っている。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['花子さんは運転手であり、終点の後も乗っていた。'] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手であり、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 ['花子さんは運転手なので、終点の後も乗っていた。'] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手なので、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: '1LNEyq0wvyq38ep',
  text: '山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。'
}] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: '1LNEyq0wvyq38ep',
  text: '山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。'
}] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'OYhEvfaWv7tqhtN',
  text: '山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。'
}] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'OYhEvfaWv7tqhtN',
  text: '山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。'
}] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'SVonKAYPEhJVnke',
  text: '山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。'
}] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'SVonKAYPEhJVnke',
  text: '山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。'
}] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'fXITmTLtCvH99MD',
  text: '山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。'
}] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{
  storyId: 'fXITmTLtCvH99MD',
  text: '山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。'
}] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{ storyId: '5jvHv0VLYZJBJv4', text: '花子さんは運転手なので、終点の後も乗っていた。' }] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手なので、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{ storyId: 'JgBouyhkhpiR0L5', text: '花子さんは運転手なので、終点の後も乗っていた。' }] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手なので、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{ storyId: 'lkeEI9sxkrD7GKk', text: '花子さんは運転手なので、終点の後も乗っていた。' }] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手なので、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [{ storyId: 'mYIsn3D5jkVOkCK', text: '花子さんは運転手なので、終点の後も乗っていた。' }] 1`] = `
{
  "distance": 0.36,
  "input": "花子さんは運転手なので、終点の後も乗っていた。",
  "result": "Covers",
  "truth": "花子さんはバスの運転手であり、花子さんにとってバスの運転席が仕事場である。花子さんは今日も仕事場である市営バスの運転席に向かい、そして業務にあたる。終点に着いた後も業務のためにバスの運転席に座り続ける必要があった。",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。] 1`] = `
{
  "distance": 0,
  "input": "山田さんは男性用トイレにいたが、他の男性に見つかりそうになり女性用トイレに逃げ込んだ。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して正しい解答をするとCoversが返る。 [山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。] 1`] = `
{
  "distance": 0.22,
  "input": "山田さんは男性用トイレにいたが、男性から隠れるために女性用トイレに移動した。",
  "result": "Covers",
  "truth": "山田さんがいるのは人気のない深夜の公園の公衆便所。 その日、山田さん人を殺害し男性用トイレに死体を隠そうとした。 しかし、その途中で山田さんはトイレに向かってくる複数の男性の喋り声に気づく。 山田さんは返り血を浴びており、もしも彼らに見つかれば間違いなく不審に思われるだろう。公園は見通しが良いためトイレから出ることも難しい。幸い、男性の位置からはトイレの入り口は見えない。 山田さんは彼らに見つからないように、男性達が入らないであろう女性用トイレに逃げ込んだ。
",
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['山田さんは人を殺害し、見つからないように女性トイレに隠れた'] 1`] = `
{
  "distance": 0.39,
  "input": "山田さんは人を殺害し、見つからないように女性トイレに隠れた",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['山田さんは女性になりたくて女性用トイレに入った'] 1`] = `
{
  "distance": 0.32,
  "input": "山田さんは女性になりたくて女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['山田さんは女性用トイレに入った'] 1`] = `
{
  "distance": 0.32,
  "input": "山田さんは女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['山田さんは覗きをするために女性用トイレに入った。'] 1`] = `
{
  "distance": 0.3,
  "input": "山田さんは覗きをするために女性用トイレに入った。",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['山田さんは隠れるために女性用トイレに入った'] 1`] = `
{
  "distance": 0.31,
  "input": "山田さんは隠れるために女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る ['花子さんは死んでしまったため、終点でもずっと座っていた。'] 1`] = `
{
  "distance": 0.43,
  "input": "花子さんは死んでしまったため、終点でもずっと座っていた。",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る [山田さんは人を殺害し、見つからないように女性トイレに隠れた] 1`] = `
{
  "distance": 0.39,
  "input": "山田さんは人を殺害し、見つからないように女性トイレに隠れた",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る [山田さんは女性用トイレに入った] 1`] = `
{
  "distance": 0.32,
  "input": "山田さんは女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る [山田さんは覗きをするために女性用トイレに入った。] 1`] = `
{
  "distance": 0.3,
  "input": "山田さんは覗きをするために女性用トイレに入った。",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る [山田さんは隠れるために女性用トイレに入った] 1`] = `
{
  "distance": 0.31,
  "input": "山田さんは隠れるために女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;

exports[`trpc/truth > 解答した内容に対して、結果が返る > 真相に対して間違えた解答をするとがWrong返る [山田は女性になりたくて女性用トイレに入った] 1`] = `
{
  "distance": 0.35,
  "input": "山田は女性になりたくて女性用トイレに入った",
  "result": "Wrong",
  "truth": null,
}
`;
